cmd/gdeploy/main.go

```diff
@@ -105,6 +101,15 @@ func WithBeforeFuncs(cmd *cli.Command, funcs ...cli.BeforeFunc) *cli.Command {
 	// e.g check if deployment config exists before checking for aws credentials
 	b := cmd.Before
 	cmd.Before = func(c *cli.Context) error {
+		args := c.Args().Slice()
+
+		// if help argument is provided then skip this check.
+		for _, arg := range args {
+			if arg == "-h" || arg == "--help" || arg == "help" {
+				return nil
+			}
+		}
```

Comment:
@JoshuaWilkes  when `beforeFunc` was blocking CLI cmds to respond with a help output. 
For cases like: 
`gdeploy notifications slack -h` 

I have added this check in `beforeFunc` not sure if there's a better way though.